lib/src/extensions/line_stream.dart

```diff
@@ -181,22 +180,38 @@ extension LineStreamExtensions on Stream<String> {
   ///
   /// If [name] is passed, it's used as the name of the spawned script.
   ///
-  /// See also [LineStreamExtensions.xargs], which takes arguments from stdin
-  /// rather than from a string stream.
+  /// [Script.kill] closes the input stream. It returns `true` if the stream was
+  /// interrupted and the script exits with [Script.exitCode] `143`, or `false`
+  /// if the stream was already closed.
+  ///
+  /// The [callback] can't be interrupted by calling [kill], but the [onSignal]
+  /// callback allows capturing those signals so the callback may react
+  /// appropriately. Calling [kill] returns `true` even without an [onSignal]
+  /// callback if the stream is interrupted.
+  ///
+  /// See also `xargs` in `package:cli_script/cli_script.dart`, which takes
+  /// arguments from [stdin] rather than from this string stream.
   Script xargs(FutureOr<void> callback(List<String> args),
-      {int? maxArgs, String? name}) {
+      {int? maxArgs, String? name, void onSignal(ProcessSignal signal)?}) {
     if (maxArgs != null && maxArgs < 1) {
       throw RangeError.range(maxArgs, 1, null, 'maxArgs');
     }
 
-    return Script.capture((stdin) async {
-      if (maxArgs == null) {
-        await callback(await toList());
-      } else {
-        await for (var slice in slices(maxArgs)) {
-          await callback(slice);
-        }
-      }
-    }, name: name ?? 'xargs');
+    var completer = Completer<int>();
+    late StreamSubscription<void> subscription;
+    return Script.capture(
+        (_) async {
+          var chunks = maxArgs != null ? slices(maxArgs) : toList().asStream();
+          subscription = chunks.asyncMap(callback).listen((_) {},
+              onDone: () => completer.complete(0), cancelOnError: true);
```

Comment:
Something to note when working with streams in the future: you should *always* add an `onError` handler. Otherwise, any error that comes through the stream will be treated as totally unhandled and will probably crash the entire application without a way for callers to handle it.

Reply:
I added a test that correctly captures the exception when the stream gets `addError` even without an `onError` handler, I guess the `Script.capture` took care of it for me here and otherwise it would have crashed as you mention, but I added it nonetheless for completeness